python/ray/experimental/state.py

```diff
@@ -345,12 +349,29 @@ def task_profiles(self):
         executions of that task. The second element is a list of profiling
         information for tasks where the events have no task ID.
     """
+    if start is None:
+      start = 0
+    if num is None:
+      num = sys.maxsize
+
     task_info = dict()
-    event_names = self.redis_client.keys("event_log*")
-    for i in range(len(event_names)):
-      event_list = self.redis_client.lrange(event_names[i], 0, -1)
+    event_log_sets = self.redis_client.keys("event_log*")
+    '''
+    The heap is used to maintain the set of x tasks that occurred the most
+    recently across all of the workers, where x is defined as the function
+    parameter num. The key is the start time of the "get_task" component of
+    each task. Calling heappop will result in the taks with the earliest
+    "get_task_start" to be removed from the heap.
+    '''
+    heap = []
+    heapq.heapify(heap)
+    heap_size = 0
+    # Parse through event logs to determine task start and end points.
+    for i in range(len(event_log_sets)):
+      event_list = self.redis_client.zrangebyscore(event_log_sets[i], min=start,
+                                                   max=end, start=start, num=num)
       for event in event_list:
-        event_dict = json.loads(event)
+        event_dict = ujson.loads(event)
```

Comment:
Is the idea that `ujson` is faster than `json`? If so, how much faster?

Also, we shouldn't use both `json` and `ujson`. If it's significantly faster, then let's just do `import ujson as json` at the top.